cli/cli.ts

```diff
@@ -0,0 +1,151 @@
+const boxen = require("boxen");
+const chalk = require("chalk");
+const figlet = require("figlet");
+const inquirer = require("inquirer");
+const yargs = require("yargs/yargs");
+const { hideBin } = require("yargs/helpers");
+
+type CommandLineArgs = Array<string>;
+
+type Options = {
+  backend: string;
+  api: string;
+  database: string;
+  auth: boolean;
+};
+
+const parseArguments = (args: CommandLineArgs) => {
+  const { argv } = yargs(hideBin(args))
+    .option("b", {
+      alias: "backend",
+      description: "Backend language",
+      choices: ["typescript", "python"],
+    })
+    .option("a", {
+      alias: "api",
+      description: "API type",
+      choices: ["rest", "graphql"],
+    })
+    .option("d", {
+      alias: "database",
+      description: "Database",
+      choices: ["postgresql", "mongodb"],
+    })
+    .option("au", {
+      alias: "auth",
+      type: "boolean",
+      description: "Include built-in auth features",
+    });
+
+  if (argv.backend === "typescript") {
+    argv.backend = "TypeScript (Node/Express)";
+  } else if (argv.backend === "python") {
+    argv.backend = "Python (Flask)";
+  }
+  if (argv.api === "rest") {
+    argv.api = "REST";
+  } else if (argv.api === "graphql") {
+    argv.api = "GraphQL";
+  }
+  if (argv.database === "postgresql") {
+    argv.database = "PostgreSQL";
+  } else if (argv.database === "mongodb") {
+    argv.database = "MongoDB";
+  }
```

Comment:
Instead of hardcoding these values everywhere let's make one object that stores the choices for each configurable field with its option id and readable name.
We can condense this section and also use that object as the source of truth.